composer.json

```diff
@@ -28,11 +28,9 @@
 	},
 	"require": {
 		"php": ">=7.1",
-		"composer/installers": "1.*,>=1.0.1",
-		"mediawiki/semantic-media-wiki": "~3.1|~4.0"
+		"composer/installers": "1.*,>=1.0.1"
```

Comment:
@thomas-topway-it Out of curiosity: How does the extension know it requires SMW hard?

Reply:
@kghbln hello Karsten, from `SemanticMetaTags.php`

 ```
	public static function onExtensionFunction() {

		if ( !defined( 'SMW_VERSION' ) ) {

			if ( PHP_SAPI === 'cli' || PHP_SAPI === 'phpdbg' ) {
				die( "\nThe 'Semantic Meta Tags' extension requires 'Semantic MediaWiki' to be installed and enabled.\n" );
			} else {
				die(
					'<b>Error:</b> The <a href="https://www.semantic-mediawiki.org/wiki/Extension:Semantic_Meta_Tags">Semantic Meta Tags</a> ' .
					'extension requires <a href="https://www.semantic-mediawiki.org/wiki/Semantic_MediaWiki">Semantic MediaWiki</a> to be ' .
					'installed and enabled.<br>'
				);
			}
		}
```

(if I understood correctly the question)

Reply:
Yeah, you did understand correctly. I was asking out of curiosity since other extensions still do it even both in https://github.com/SemanticMediaWiki/SemanticExtraSpecialProperties/blob/master/extension.json#L18 and https://github.com/SemanticMediaWiki/SemanticExtraSpecialProperties/blob/master/composer.json#L48 though that particular extension does no longer have a SemanticExtraSpecialProperties.php to hold this information.

Reply:
the main reason I've removed it, it is for testing purposes to allow installing the required libraries (mainly easyRdf and json-ld, see here https://github.com/SemanticMediaWiki/SemanticMetaTags/pull/80 ) in the extension's folder. Therefore once that https://github.com/SemanticMediaWiki/SemanticMetaTags/pull/80 will be merged the SMW require in composer could be restored back. (if deemed necessary, for this actually we should check the strategy followed by all similar extensions dependent on SMW)